pandas/io/formats/format.py

```diff
@@ -1444,8 +1449,19 @@ def get_result_as_array(self) -> np.ndarray:
         Returns the float values converted into strings using
         the parameters given at initialisation, as a numpy array
         """
+
+        def format_with_na_rep(values, formatter, na_rep):
+            mask = isna(values)
+            formatted = np.array(
+                [
```

Comment:
can you use this function more generally? (e.g. maybe define it in the module); can be a followup as well.

Reply:
It seems similar patterns occur elsewhere, although most are at the scalar level